actionpack/CHANGELOG.md

```diff
@@ -1,5 +1,17 @@
+*   Change `ActionDispatch::Testing::TestResponse#parsed_body` to parse HTML as
+    a Nokogiri document
+
+    ```ruby
+    get "/posts"
+    response.content_type         # => "text/html; charset=utf-8"
+    response.parsed_body.class    # => Nokogiri::HTML5::Document
+    response.parsed_body.to_html  # => "<!DOCTYPE html>\n<html>\n..."
```

Comment:
Does this mean that people that were expecting strings will have to rewrite their tests?

Reply:
Yes, it would be a breaking change.

> people that were expecting strings will have to rewrite their tests?

I wasn't sure about this. Since `response.parsed_body` and `response.body` return the same type (`String`), would they be motivated to use `response.parsed_body`?

Would upgrading from 7.0 to 7.1 be an appropriate time to introduce a breaking change? Would a breaking change require introducing a configuration? Something like `action_dispatch.html_parsed_body_as_document = true` that `<= 7.0` applications would set to `false`?

Reply:
I think it is fine to just explicitly document that the return type of that method is going to change.